src/podutils.rs

```diff
@@ -93,46 +96,229 @@ where
+    matches!((pod.spec.as_ref(), node.metadata.name.as_ref()),
+        (
+            Some(PodSpec { node_name: Some(ref pod_node_name), ..}, ..),
+            Some(node_node_name),
+        ) if pod_node_name == node_node_name
+    )
+}
+
+/// This method checks if a Pod contains all required labels including an optional value per label.
+///
+/// # Arguments
+///
+/// - `pod` - the Pod to check for labels
+/// - `required_labels` - is a BTreeMap of label keys to an optional value
+///
+/// # Example
+///
+/// ```
+/// use stackable_operator::podutils;
+/// # use k8s_openapi::api::core::v1::{Pod, PodSpec};
+/// # use k8s_openapi::apimachinery::pkg::apis::meta::v1::ObjectMeta;
+/// use std::collections::BTreeMap;
+///
+/// # let pod = Pod {
+/// #            metadata: ObjectMeta {
+/// #                ..ObjectMeta::default()
+/// #            },
+/// #            spec: None,
+/// #            status: None,
+/// #        };
+///
+/// let mut required_labels = BTreeMap::new();
+/// required_labels.insert("foo".to_string(), Some("bar".to_string()));
+///
+/// assert!(!podutils::pod_matches_labels(&pod, &required_labels));
+/// ```
+pub fn pod_matches_labels(pod: &Pod, required_labels: &BTreeMap<String, Option<String>>) -> bool {
+    // We convert the `required_labels` into a form that can be understood by `pod_matches_multiple_label_values`
+    let converted = required_labels
+        .iter()
+        .map(|(key, value)| {
+            (
+                key.clone(),
+                value.as_ref().map(|string| vec![string.clone()]),
+            )
+        })
+        .collect::<BTreeMap<_, _>>();
+    pod_matches_multiple_label_values(pod, &converted)
+}
+
+/// This method checks if a Pod contains all required labels including an optional check for values.
+///
+/// # Arguments
+///
+/// * `pod` - the Pod to check for labels
+/// * `required_labels` - is a BTreeMap of label keys to an optional vector of label values.
+///                       Multiple values can be passed in but the Pod must obviously match
+///                       _any_ of the values to be accepted
+///
+/// # Example
+///
+/// ```
+/// use stackable_operator::podutils;
+/// # use k8s_openapi::api::core::v1::{Pod, PodSpec};
+/// # use k8s_openapi::apimachinery::pkg::apis::meta::v1::ObjectMeta;
+/// use std::collections::BTreeMap;
+///
+/// # let pod = Pod {
+/// #            metadata: ObjectMeta {
+/// #                ..ObjectMeta::default()
+/// #            },
+/// #            spec: None,
+/// #            status: None,
+/// #        };
+///
+/// let mut required_labels = BTreeMap::new();
+/// required_labels.insert("foo".to_string(), None);
+///
+/// assert!(!podutils::pod_matches_multiple_label_values(&pod, &required_labels));
+/// ```
+pub fn pod_matches_multiple_label_values(
+    pod: &Pod,
+    required_labels: &BTreeMap<String, Option<Vec<String>>>,
+) -> bool {
+    let pod_labels = &pod.metadata.labels;
```

Comment:
I've thought about this function for a little while and can't shake the feeling that we could do 'better' with a two step .iter().filter(...) approach to first check for missing keys and then for matching values.  
That would also allow to give a full report of the non-matching labels in the debug message instead of just the first non-matching item.

That being said, this is not something we should spend time on currently!

Reply:
I'm not sure what you mean by "validate method", but still think we shouldn't spend time on this at the moment :)